File: geofence-api/src/main/scala/com/simacan/jfall/geofence/api/http/geofence/GeofenceTools.scala
package com.simacan.jfall.geofence.api.http.geofence

import com.simacan.jfall.geofence.api.GeofenceAPI.Coordinates

object GeofenceTools {

  //This object describes a North oriented, square shaped geofence,
  //thus both upper corners are on the same lat, bottom corners are on the same lat,
  //both left corners on the same lon and both right corners on the same lon.
  case class NorthOrientedSquareGeofence(cornerUpLeft: Coordinates, cornerBottomRight: Coordinates)

  private lazy val northOrientedSquareGeofence: NorthOrientedSquareGeofence =
    GeofenceTools.NorthOrientedSquareGeofence(
      Coordinates(52.00702321126474, 5.6270599365234375),
      Coordinates(52.027729487209015, 5.6667137145996085))

  //This method checks if the given coordinates lie within the northOrientedSquareGeofence
  def checkOnNorthOrientedSquareGeofence(coordinates: Coordinates): Boolean = {
    coordinates.lat >= northOrientedSquareGeofence.cornerUpLeft.lat &&
      coordinates.lat <= northOrientedSquareGeofence.cornerBottomRight.lat &&
      coordinates.lon >= northOrientedSquareGeofence.cornerUpLeft.lon &&
      coordinates.lon <= northOrientedSquareGeofence.cornerBottomRight.lon
  }
}

File: java/geofenceapi/src/main/java/GeofenceTools.java


class GeofenceTools {

    //This object describes a North oriented, square shaped geofence,
    //thus both upper corners are on the same lat, bottom corners are on the same lat,
    //both left corners on the same lon and both right corners on the same lon
    private static class NorthOrientedSquareGeofence {
        private Coordinates cornerUpLeft;
        private Coordinates cornerBottomRight;

        NorthOrientedSquareGeofence(Coordinates cornerUpLeft, Coordinates cornerBottomRight) {
            this.cornerUpLeft = cornerUpLeft;
            this.cornerBottomRight = cornerBottomRight;
        }
    }

    private static NorthOrientedSquareGeofence northOrientedSquareGeofence = new NorthOrientedSquareGeofence(
            new Coordinates(52.00702321126474, 5.6270599365234375),
            new Coordinates(52.027729487209015, 5.6667137145996085));

    //This method checks if the given coordinates lie within the northOrientedSquareGeofence
    static boolean checkOnNorthOrientedSquareGeofence(Coordinates coordinates) {
        return coordinates.getLat() >= northOrientedSquareGeofence.cornerUpLeft.getLat() &&
                coordinates.getLat() <= northOrientedSquareGeofence.cornerBottomRight.getLat() &&
                coordinates.getLon() >= northOrientedSquareGeofence.cornerUpLeft.getLon() &&
                coordinates.getLon() <= northOrientedSquareGeofence.cornerBottomRight.getLon();
    }
}
